web/handler.go

```diff
@@ -67,19 +69,23 @@ func NewHandler(urlStr, prefix string) (http.Handler, error) {
 		extraScripts = []string{"vendor.js"}
 	}
 
-	var buf bytes.Buffer
-	err := indexTmpl.Execute(&buf, renderData{
-		Prefix:       prefix,
-		ExtraScripts: extraScripts,
-	})
-	if err != nil {
-		return nil, err
-	}
-	h := sha256.New()
-	h.Write(buf.Bytes())
-	indexETag := fmt.Sprintf(`"sha256-%s"`, hex.EncodeToString(h.Sum(nil)))
-
 	mux.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+		cfg := config.FromContext(req.Context())
+
+		var buf bytes.Buffer
+		err := indexTmpl.Execute(&buf, renderData{
+			ApplicationName: cfg.ApplicationName(),
+			Prefix:          prefix,
+			ExtraScripts:    extraScripts,
+		})
+		if err != nil {
+			log.Log(req.Context(), err)
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
```

Comment:
This needs to return, we can only reply once to a single request.

```suggestion
		if errutil.HTTPError(ctx, w, err) {
			return
```